src/legacy/core_plugins/kibana/public/discover/np_ready/angular/context.js

```diff
@@ -68,53 +66,73 @@ getAngularModule().config($routeProvider => {
     });
 });
 
-function ContextAppRouteController(
-  $routeParams,
-  $scope,
-  AppState,
-  config,
-  $route,
-  getAppState,
-  globalState
-) {
+function ContextAppRouteController($routeParams, $scope, config, $route) {
   const filterManager = getServices().filterManager;
-  const filterStateManager = new FilterStateManager(globalState, getAppState, filterManager);
   const indexPattern = $route.current.locals.indexPattern.ip;
+  const { start, stop, appState, globalState, getAppFilters, getGlobalFilters } = getState(
```

Comment:
Would it be more obvious to give readable names for `start` & `stop` ?

Reply:
definitely! and I've seen a nifty PR targeting Visualize, that does it right, hope the author don't mind if I copy&paste, `startStateSync`, `stopStateSync`

Reply:
hehe  😂 
Hope that author don't have strict license rules  😂